internal/app/secretless/handlers/pg/backend_config_test.go

```diff
@@ -0,0 +1,118 @@
+package pg
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBackendConfigCorrectlyParsesOptions(t *testing.T) {
+	options := make(map[string][]byte)
+
+	options["host"] = []byte("myhost")
+	options["port"] = []byte("1234")
+	options["username"] = []byte("myusername")
+	options["password"] = []byte("mypassword")
+
+	expectedBackendConfig := BackendConfig{
+		Host:         "myhost",
+		Port:         "1234",
+		Username:     "myusername",
+		Password:     "mypassword",
+		Options:      make(map[string]string),
+		QueryStrings: map[string]string{},
+	}
+
+	actualBackendConfig, err := NewBackendConfig(options)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedBackendConfig, *actualBackendConfig)
+}
+
+func TestBackendConfigCorrectlyIsolatesSSLOptions(t *testing.T) {
+	options := make(map[string][]byte)
+
+	options["host"] = []byte("myhost")
+	options["port"] = []byte("1234")
+	options["username"] = []byte("myusername")
+	options["password"] = []byte("mypassword")
+
+	options["sslrootcert"] = []byte("mysslrootcert")
+	options["sslmode"] = []byte("mysslmode")
+	options["sslkey"] = []byte("mysslkey")
+	options["sslcert"] = []byte("mysslcert")
+
+	expectedBackendConfig := BackendConfig{
+		Host:     "myhost",
+		Port:     "1234",
+		Username: "myusername",
+		Password: "mypassword",
+		Options:  make(map[string]string),
+		QueryStrings: map[string]string{
+			"sslrootcert": "mysslrootcert",
+			"sslmode":     "mysslmode",
+			"sslkey":      "mysslkey",
+			"sslcert":     "mysslcert",
+		},
+	}
+
+	actualBackendConfig, err := NewBackendConfig(options)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedBackendConfig, *actualBackendConfig)
+}
+
+func TestPortDefaultsTo5432IfNotProvided(t *testing.T) {
+	options := make(map[string][]byte)
+
+	options["host"] = []byte("myhost")
+	options["username"] = []byte("myusername")
+	options["password"] = []byte("mypassword")
+
+	expectedBackendConfig := BackendConfig{
+		Host:         "myhost",
+		Port:         "5432",
+		Username:     "myusername",
+		Password:     "mypassword",
+		Options:      make(map[string]string),
+		QueryStrings: map[string]string{},
+	}
+
+	actualBackendConfig, err := NewBackendConfig(options)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedBackendConfig, *actualBackendConfig)
+}
+
+func TestBackendConfigAddsUnknownCredentialsToOptions(t *testing.T) {
+	options := make(map[string][]byte)
+
+	options["host"] = []byte("myhost")
+	options["port"] = []byte("1234")
+	options["foo"] = []byte("3306")
+	options["username"] = []byte("myusername")
+	options["bar"] = []byte("data")
+	options["password"] = []byte("mypassword")
+
+	expectedOptions := map[string]string{
+		"foo": "3306",
+		"bar": "data",
+	}
+
+	actualBackendConfig, err := NewBackendConfig(options)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedOptions, (*actualBackendConfig).Options)
+}
+
+func TestBackendConfigCanUseAddressforConfiguration(t *testing.T) {
```

Comment:
name change: `TestAddressCanBeUsedInsteadOfHostAndPort`
my initial suggestion was `TestAddressIsSupported`.  Normally I much prefer shorter names but in this case the clarity might justify the verbosity.